tests/unittests/BackendTestUtils2.cpp

```diff
@@ -401,38 +418,52 @@ void trainLocalResponseNormalizationNet(Tensor *inputs, Tensor *weights,
   TC.learningRate = 0.06;
   TC.momentum = 0.1;
   TC.L2Decay = 0.01;
-  auto &mod = EE.getModule();
-  Function *F = mod.createFunction("main");
-  auto *var1 = createPlaceholder(mod, bindings, inputs, "var1");
-  auto *var2 = createPlaceholder(mod, bindings, selected, "var2");
-  auto *fc = F->createFullyConnected(bindings, "fc", var1, bias->dims()[0]);
-  bindings.get(cast<Placeholder>(fc->getWeights()))->assign(weights);
-  bindings.get(cast<Placeholder>(fc->getBias()))->assign(bias);
-  auto *reshape1 = F->createReshape("reshape1", fc, shape1);
-  auto *lrn =
-      F->createLocalResponseNormalization("lrn", reshape1, 2, 2.0, 0.5, 1.0);
-  auto *reshape2 = F->createReshape("reshape2", lrn, shape2);
-  auto *softmax = F->createSoftMax("softmax", reshape2, var2);
-  auto *result = F->createSave("ret", softmax);
-  auto *resultTensor = bindings.allocate(result->getPlaceholder());
-
-  Function *TF = glow::differentiate(F, TC);
-  EE.compile(CompilationMode::Train, TF);
-  runBatch(EE, bindings, 8, sampleCounter, {var1, var2}, {inputs, selected});
-
-  EE.compile(CompilationMode::Infer, F);
-
-  runBatch(EE, bindings, 1, sampleCounter, {var1, var2}, {inputs, selected});
-  out->assign(resultTensor);
+  Placeholder *var1, *var2;
+  std::string fName;
+  for (auto *EE : engines) {
+    auto &mod = EE->getModule();
+    Function *F = mod.createFunction("main");
+    fName = F->getName();
+    var1 = createPlaceholder(mod, bindings, inputs, "var1");
+    var2 = createPlaceholder(mod, bindings, selected, "var2");
+    auto *fc = F->createFullyConnected(bindings, "fc", var1, bias->dims()[0]);
+    bindings.get(cast<Placeholder>(fc->getWeights()))->assign(weights);
+    bindings.get(cast<Placeholder>(fc->getBias()))->assign(bias);
+    auto *reshape1 = F->createReshape("reshape1", fc, shape1);
+    auto *lrn =
+        F->createLocalResponseNormalization("lrn", reshape1, 2, 2.0, 0.5, 1.0);
+    auto *reshape2 = F->createReshape("reshape2", lrn, shape2);
+    auto *softmax = F->createSoftMax("softmax", reshape2, var2);
+    auto *result = F->createSave("ret", softmax);
+    bindings.allocate(result->getPlaceholder());
+  }
+  trainingBindings.allocate(EET.getModule().getPlaceholders());
+  bindings.copyTrainedWeightsTo(trainingBindings);
+  bindings.clear();
+  bindings.allocate(EEI.getModule().getPlaceholders());
+  auto *TF = glow::differentiate(EET.getModule().getFunction(fName), TC);
+  auto tfName = TF->getName();
+  EET.compile(CompilationMode::Train);
```

Comment:
tiny nit: new line after here to try to separate the different phases. Also consistent with other places.